File: java/1859.java
import java.util.Arrays;

class Solution {
    public String sortSentence(String s) {
        String[] sentenceArray = s.split(" ");
        String[] resultArray = new String[sentenceArray.length];

        for(String word: sentenceArray){
            int thisIndex = Character.getNumericValue(word.charAt(word.length() - 1));
            resultArray[thisIndex - 1] = word.substring(0, word.length() - 1);
        }

        return String.join(" ", resultArray);
    }
}

class Main {
    public static void main(String[] args) {
        Solution solution = new Solution();

        String s1 = "is2 sentence4 This1 a3";
        String r1 = solution.sortSentence(s1);
        System.out.println(r1);
    }
}

File: kotlin/1859.kt
class Solution {
    fun sortSentence(s: String): String {
        val inputStringAsArray = s.split(" ").toTypedArray()
        val returnStringArray = inputStringAsArray.copyOf()

        for(i in inputStringAsArray) {
            returnStringArray[i.last().toString().toInt() - 1] = i.dropLast(1)
        }

        return returnStringArray.joinToString(" ")
    }
}

fun main(){
    val solution = Solution()

    val input1 = "is2 sentence4 This1 a3"
    val input2 = "Myself2 Me1 I4 and3"

    println(solution.sortSentence(input1))
    println(solution.sortSentence(input2))
}